### PR TITLE
Discretizer ameva and small fixes

### PR DESCRIPTION
**Pull request by MarvinBleier**
Added a second supervised discretizer, Ameva. Unlike the FUSINTER discretization, this algorithm uses a variation of the contingency coefficents. Also Ameva starts with 0 cut points and tentatively tests if an additional cut point would increase the coefficent.

Small refactoring of Interval and AbstractDiscretizer were needed. Also small bugs in FUSINTER were fixed.

### CODE REVIEW

File: DefaultConfigsExtension/src/main/java/de/viadee/xai/anchor/adapter/tabular/discretizer/Interval.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        return new DiscretizationTransition(new NumericDiscretizationOrigin(
+                keyValuePairs.get(begin).getKey(),
+                keyValuePairs.get(end).getKey()),
+                medianIndexValue()
+        );
+    }
+
+    private double medianIndexValue() {
+        if (getSize() % 2 == 0) {
+            return (keyValuePairs.get(end + 1 - getSize() / 2).getKey().doubleValue() + keyValuePairs.get(end + 1 - getSize() / 2 - 1).getKey().doubleValue()) / 2;
```
Comment (fkoehne) [inline DefaultConfigsExtension/src/main/java/de/viadee/xai/anchor/adapter/tabular/discretizer/Interval.java]: getKey().doubleValue() removes the flexibility you created by using number as a a Key. I would suggest to keep things simple and go for a double datastructure.

File: DefaultConfigsExtension/src/main/java/de/viadee/xai/anchor/adapter/tabular/discretizer/impl/AmevaDiscretizer.java [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+        bCutPoints = IntStream.range(1, values.length)
+                .mapToObj(i -> ((Double) values[i] + (Double) values[i-1] )/ 2)
+                .sorted().distinct()
+                .collect(Collectors.toList());
+
+        double globalAmeva = 0.0;
+        double ameva = createNewCutPoint(globalAmeva);
+
+        while(ameva > globalAmeva){
```
Comment (fkoehne) [inline DefaultConfigsExtension/src/main/java/de/viadee/xai/anchor/adapter/tabular/discretizer/impl/AmevaDiscretizer.java]: Is this save to do? A code comment would be appreciated.

**Comment by fkoehne** [comment]
Nice work. It looks like you had a conversation with sonar already and that really pays off.

File: DefaultConfigsExtension/src/main/java/de/viadee/xai/anchor/adapter/tabular/discretizer/impl/AmevaDiscretizer.java (hunk 2 above)
Comment (MarvinBleier) [reply DefaultConfigsExtension/src/main/java/de/viadee/xai/anchor/adapter/tabular/discretizer/impl/AmevaDiscretizer.java]: Added a second exit-condition, which terminites the loop, if the amount of potential cut points is zero.

File: DefaultConfigsExtension/src/main/java/de/viadee/xai/anchor/adapter/tabular/discretizer/Interval.java (hunk 1 above)
Comment (MarvinBleier) [reply DefaultConfigsExtension/src/main/java/de/viadee/xai/anchor/adapter/tabular/discretizer/Interval.java]: Changed Interval and all supervised Discretizer AbstractMaps to <Double, Double>. 
Tested, and it still works with the NumericDiscretizerOrigin.

**Comment by fkoehne** [comment]
Last fail was a network glitch - sonar is happy with the changes.

**Review by TobiasGoerke — APPROVED** [review]

**Pull request merged**